src/ray/object_manager/plasma/create_request_queue.cc

```diff
@@ -173,12 +196,31 @@ Status CreateRequestQueue::ProcessRequests() {
         oom_start_time_ns_ = now;
       }
 
-      // Notify the scheduler that object creation is blocked at this priority.
-      on_object_creation_blocked_callback_(queue_it->first.first);
-      if (!should_spill_) {
-        RAY_LOG(INFO) << "Object creation of priority " << queue_it->first.first << " blocked";
-        return Status::TransientObjectStoreFull("Waiting for higher priority tasks to finish");
-      }
+	  if(RayConfig::instance().enable_BlockandEvictTasks()){
+		on_object_creation_blocked_callback_(queue_it->first.first);
```

Comment:
Instead of having two separate callbacks, can we do one callback that sets the blocked priority and a flag for whether to also evict tasks?

I think this will be cleaner and also easier to debug. Since the callbacks here are concurrent, it's going to be hard to figure out what's going on if there are too many callbacks.